Modules/_fuzz/fuzzer.cpp

```diff
@@ -0,0 +1,118 @@
+
+#include <Python.h>
+#include <stdlib.h>
+#include <inttypes.h>
+
+// Fuzz PyFloat_FromString as a proxy for float(str).
+static int fuzz_builtin_float(const char* data, size_t size) {
+    PyObject* s = PyBytes_FromStringAndSize(data, size);
+    if (s == NULL) return 0;
+    PyObject* f = PyFloat_FromString(s);
+    if (PyErr_Occurred() && PyErr_ExceptionMatches(PyExc_ValueError)) {
+        PyErr_Clear();
+    }
+
+    Py_XDECREF(f);
+    Py_DECREF(s);
+    return 0;
+}
+
+// Fuzz PyLong_FromUnicodeObject as a proxy for int(str).
+static int fuzz_builtin_int(const char* data, size_t size) {
+    int base = _Py_HashBytes(data, size) % 36;
+    if (base == 1) {
+        base = 0;
+    }
+    if (base == -1) {
+        return 0;  // An error occurred, bail early.
+    }
+    if (base < 0) {
+        base = -base;
+    }
+
+    PyObject* s = PyUnicode_FromStringAndSize(data, size);
+    if (PyErr_Occurred()) {
+        if (PyErr_ExceptionMatches(PyExc_UnicodeDecodeError)) {
+            PyErr_Clear();
+        }
+        return 0;
+    }
+    PyObject* l = PyLong_FromUnicodeObject(s, base);
+    if (PyErr_Occurred() && PyErr_ExceptionMatches(PyExc_ValueError)) {
+        PyErr_Clear();
+    }
+    PyErr_Clear();
+    Py_XDECREF(l);
+    Py_DECREF(s);
+    return 0;
+}
+
+// Fuzz PyUnicode_FromStringAndSize as a proxy for unicode(str).
+static int fuzz_builtin_unicode(const char* data, size_t size) {
+    PyObject* s = PyUnicode_FromStringAndSize(data, size);
+    if (PyErr_Occurred() && PyErr_ExceptionMatches(PyExc_UnicodeDecodeError)) {
+        PyErr_Clear();
+    }
+    Py_XDECREF(s);
+    return 0;
+}
+
+// Run fuzzer and abort on failure.
+static int _run_fuzz(const uint8_t *data, size_t size, int(*fuzzer)(const char* , size_t)) {
+    int rv = fuzzer("", 0);
+    if (PyErr_Occurred()) {
+        // Fuzz tests should handle expected errors for themselves.
+        PyErr_Print();
+        abort();
+    }
+    // Someday the return value might mean something, propagate it.
+    return rv;
+}
+
+// CPython generates a lot of leak warnings for whatever reason.
+extern "C" int __lsan_is_turned_off(void) { return 1; }
+
+// Fuzz test interface.
+// This returns the bitwise or of all fuzz test's return values.
+//
+// All fuzz tests must return 0, as all nonzero return codes are reserved for
+// future use -- we propagate the return values for that future case.
+// (And we bitwise or when running multiple tests to verify that normally we
+// only return 0.)
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+    if (!Py_IsInitialized()) {
+        // LLVMFuzzerTestOneInput is called repeatedly from the same process, with
+        // no separate initialization phase, sadly, so we need to initialize CPython
+        // ourselves on the first run.
+        Py_InitializeEx(0);
+    }
+
+    int rv = 0;
+
+#define _Py_FUZZ_YES(test_name) (defined(_Py_FUZZ_##test_name) || !defined(_Py_FUZZ_ONE))
```

Comment:
I don't think I understand the suggestion. Care to elaborate a bit?

It might be worth looking at commit  d112252 to see how test discover was when I first hacked something together. I couldn't find any solution I liked, but I'm no C expert. (In Python this would be no sweat!)